apis/v1alpha1/archive.go

```diff
@@ -0,0 +1,91 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package v1alpha1
+
+import (
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ArchiveSpec defines the desired state of Archive.
+//
+// An Archive object that contains details about an archive.
+type ArchiveSpec struct {
+
+	// The name for the archive to create.
+	// +kubebuilder:validation:Required
+	ArchiveName *string `json:"archiveName"`
```

Comment:
nit: Should we rename this field to `Name` ? I didn't check the other resources by ideally (before going GA) we need to reduce redundant names like `EventBusName` etc...

Reply:
I'm torn on those because customers might expect the actual CF or Terraform field names which I guess use the model names. We had this discussion w/ @RedbackThomson on the `Rule` target parameters as well. Would have to change all the other resources too. But rather now than later depending on what we decide.

Reply:
Can you please provide me with a `generator.yaml` example for how to perform these renames? I found a `renamesConfig` object and looking at how S3 uses it I'm not sure if:

- this needs to be done per resource and its related CRUD operations
- how it affects `Ref` fields

Reply:
Here is an example of how to rename `FunctionName` to `Name` in lambda-controller: https://github.com/aws-controllers-k8s/lambda-controller/blob/main/generator.yaml#L64-L74 (This example targets all CRUD operations)

`references` and `is_immutable` fields should not be impacted. In the case they are not functioning correctly we can submit a fix to the code-generator.

Reply:
Discussed with @A-Hilaly that we'll do renames PR once all resources are in